### PR TITLE
Migrate to aiogram 3 & make role middleware

### PR DESCRIPTION
**Pull request by Ninnjah**
Migrate bot to aiogram 3
Add role middleware for check user is admin
Changed the bot structure

### CODE REVIEW

File: frontends/telegram/app/bot.py [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
 async def main() -> None:
     """
     Entry point of the bot, should be ran with Docker
     """
+    token = Settings().subscriby_telegram_token
+    bot = Bot(token=token, parse_mode="html")
+    dp = Dispatcher(storage=MemoryStorage())
+
+    dp.message.outer_middleware(RoleMiddleware(admin_list=Settings().subscriby_telegram_admin_ids))
+    dp.callback_query.outer_middleware(RoleMiddleware(admin_list=Settings().subscriby_telegram_admin_ids))
```
Comment (kirillzhosul) [inline frontends/telegram/app/bot.py:21]: May be better when there is one instance of the middleware

File: frontends/telegram/app/handlers/admin/subscriptions.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+from frontends.telegram.app.states import PublishSubscription
+from frontends.telegram.app.settings import Settings
+from frontends.telegram.app.keyboards.reply.admin import main_kb
+from frontends.telegram.app.filters.role import RoleFilter
+from frontends.telegram.app.models.role import UserRole
+
+router = Router(name=__name__)
+# Apply RoleFilter to all router's handlers
+router.message.filter(RoleFilter(UserRole.ADMIN))
+router.callback_query.filter(RoleFilter(UserRole.ADMIN))
```
Comment (kirillzhosul) [inline frontends/telegram/app/handlers/admin/subscriptions.py:17]: Same may be here

File: frontends/telegram/app/handlers/admin/subscriptions.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+
+def _api_call(method: str, params: dict) -> dict | None:
+    """
```
Comment (kirillzhosul) [inline frontends/telegram/app/handlers/admin/subscriptions.py:21]: Later should be migrated inside service / business logic

File: frontends/telegram/app/handlers/user/start.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+router = Router(name=__name__)
+
+
+# Apply RoleFilter to only this handler
+@router.message(CommandStart())
+async def start_command(message: Message) -> None:
+    """
+    Handler for `/start` command.
+    """
+    await message.answer("<b>Sorry, you are not an admin!</b>")
```
Comment (kirillzhosul) [inline frontends/telegram/app/handlers/user/start.py:14]: Does that even check for admin?

**Review by kirillzhosul — APPROVED** [review]

**Comment by kirillzhosul** [comment]
LGTM, Thanks you! Would be nice if you request some new features or etc, currently Telegram bot is not widely used

**Comment by kirillzhosul** [comment]
Close #2 
Close #3

File: frontends/telegram/app/handlers/user/start.py (hunk 4 above)
Comment (Ninnjah) [reply frontends/telegram/app/handlers/user/start.py:14]: No. This handler for non admin users. Simple send them message

File: frontends/telegram/app/bot.py (hunk 1 above)
Comment (Ninnjah) [reply frontends/telegram/app/bot.py:21]: aiogram 3 has a new middlewares that applied for messages and callbacks separately.
We can left only message middlewares, but if we add callbacks for this bot we'll add callback middleware again

File: frontends/telegram/app/handlers/admin/subscriptions.py (hunk 2 above)
Comment (Ninnjah) [reply frontends/telegram/app/handlers/admin/subscriptions.py:17]: Here we apply filter for events, no middlewares. And filters applies for messages and callbacks separately just like middlewares.
We can also apply filters to each handler using a decorator, but it is much easier to apply filters to the entire router.

File: frontends/telegram/app/handlers/admin/subscriptions.py (hunk 3 above)
Comment (Ninnjah) [reply frontends/telegram/app/handlers/admin/subscriptions.py:21]: No problem. When will it be migrated we can just update logic for this handler

**Pull request merged**